core/lib/da_client/src/clients/gcs/mod.rs

```diff
@@ -0,0 +1,61 @@
+use std::{
+    fmt,
+    fmt::{Debug, Formatter},
+    sync::Arc,
+};
+
+use async_trait::async_trait;
+use zksync_config::ObjectStoreConfig;
+use zksync_da_layers::{
+    types::{DataAvailabilityError, DispatchResponse, InclusionData},
+    DataAvailabilityInterface,
+};
+use zksync_object_store::{ObjectStore, ObjectStoreFactory};
+use zksync_types::{pubdata_da::StorablePubdata, L1BatchNumber};
+
+pub(crate) struct GCSDAClient {
+    object_store: Arc<dyn ObjectStore>,
+}
+
+impl GCSDAClient {
+    pub async fn new(object_store_conf: ObjectStoreConfig) -> Self {
+        GCSDAClient {
+            object_store: ObjectStoreFactory::create_from_config(&object_store_conf).await,
+        }
+    }
+}
+
+#[async_trait]
+impl DataAvailabilityInterface for GCSDAClient {
+    async fn dispatch_blob(
+        &self,
+        batch_number: u32,
+        data: Vec<u8>,
```

Comment:
what should happen if 'data' is too large for a given DA?

Reply:
it will never be dispatched and manual intervention will be required

generally, that's the responsibility of the operator to set the batch pubdata limit to be lower than the limitations of the DA layer (and their blob sizes are almost 2MB, so that's unlikely), but we can have some kind of a safeguard that checks that the batch pubdata limit is lower than the max blob size for specific DA layer

Reply:
added the safeguard for that